src/treetensornetworks/opsum_to_ttn.jl

```diff
@@ -424,212 +579,62 @@ function Base.isless(a1::ArrElem{T,N}, a2::ArrElem{T,N})::Bool where {T,N}
+    if a1.idxs[n] != a2.idxs[n]
+      return a1.idxs[n] < a2.idxs[n]
     end
-    sparseTTN[v] = T
   end
-  return sparseTTN, edge_orders
+  return a1.val < a2.val
 end
 
-"""
-    fsmTTN(os::OpSum{C}, sites::IndsNetwork{V,<:Index}, root_vertex::V, kwargs...) where {C,V}
+# 
+# Sparse finite state machine construction
+# 
 
-Construct a dense TreeTensorNetwork from sparse finite state machine
-represenatation, without compression.
-"""
-function fsmTTN(
-  os::OpSum{C}, sites::IndsNetwork{V,<:Index}, root_vertex::V; trunc=false, kwargs...
-)::TTN where {C,V}
-  ValType = ITensors.determineValType(ITensors.terms(os))
-  # start from finite state machine
-  fsm, edge_orders = finite_state_machine(os, sites, root_vertex)
-  # some trickery to get link dimension for every edge
-  link_space = Dict{edgetype(sites),Index}()
-  function get_linkind!(link_space, e)
-    if !haskey(link_space, e)
-      d = findfirst(x -> (x == e || x == reverse(e)), edge_orders[src(e)])
-      link_space[e] = Index(size(fsm[src(e)], d), edge_tag(e))
-    end
-    return link_space[e]
-  end
-  # compress finite state machine into dense form
-  H = TTN(sites)
-  for v in vertices(sites)
-    linkinds = [get_linkind!(link_space, e) for e in edge_orders[v]]
-    linkdims = dim.(linkinds)
-    H[v] = ITensor()
-    for (T_ind, t) in nonzero_pairs(fsm[v])
-      any(map(x -> abs(coefficient(x)) > eps(), t)) || continue
-      T = zeros(ValType, linkdims...)
-      T[T_ind] += one(ValType)
-      T = itensor(T, linkinds)
-      H[v] += T * computeSiteSum(sites, t)
-    end
-  end
-  # add option for numerical truncation, but throw warning as this can fail sometimes
-  if trunc
-    @warn "Naive numerical truncation of TTN Hamiltonian may fail for larger systems."
-    # see https://github.com/ITensor/ITensors.jl/issues/526
-    lognormT = lognorm(H)
-    H /= exp(lognormT / nv(H)) # TODO: fix broadcasting for in-place assignment
-    H = truncate(H; root_vertex, kwargs...)
-    H *= exp(lognormT / nv(H))
-  end
-  return H
+# allow sparse arrays with ITensors.Sum entries
+function Base.zero(::Type{S}) where {S<:Sum}
+  return S()
 end
+Base.zero(t::Sum) = zero(typeof(t))
+
 
+#=
 function computeSiteSum(
   sites::IndsNetwork{V,<:Index}, ops::Sum{Scaled{C,Prod{Op}}}
 )::ITensor where {V,C}
-  ValType = ITensors.determineValType(ITensors.terms(ops))
+  coefficient_type = ITensors.determineValType(ITensors.terms(ops))
   v = ITensors.site(ITensors.argument(ops[1])[1])
   T =
-    convert(ValType, coefficient(ops[1])) *
+    convert(coefficient_type, coefficient(ops[1])) *
     computeSiteProd(sites, ITensors.argument(ops[1]))
   for j in 2:length(ops)
     (ITensors.site(ITensors.argument(ops[j])[1]) != v) &&
       error("Mismatch of vertex labels in computeSiteSum")
     T +=
-      convert(ValType, coefficient(ops[j])) *
+      convert(coefficient_type, coefficient(ops[j])) *
       computeSiteProd(sites, ITensors.argument(ops[j]))
   end
   return T
 end
+=#
```

Comment:
Seems better to delete this instead of commenting it out since it was moved to `deprecated_opsum_to_ttn.jl`.